bin/node/src/service.rs

```diff
@@ -294,3 +295,102 @@ pub fn new_full(
     network_starter.start_network();
     Ok(task_manager)
 }
+
+pub fn new_nonvalidator(
```

Comment:
I guess the difference from `new_full` is adding Validator set to network proposer factory and starting aura. So maybe we could have less code duplication by extracting those things to a separate function and calling them depending on node type?

Reply:
done